pkg/asset/agent/image/baseiso.go

```diff
@@ -0,0 +1,141 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// BaseIso generates the base ISO file for the image
+type BaseIso struct {
+	File *asset.File
+}
+
+var (
+	baseIsoFilename = ""
+)
+
+var _ asset.WritableAsset = (*BaseIso)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *BaseIso) Name() string {
+	return "BaseIso"
+}
+
+// getIsoFile is a pluggable function that gets the base ISO file
+type getIsoFile func(config *types.InstallConfig) (string, error)
+
+type getIso struct {
+	getter getIsoFile
+}
+
+func newGetIso(getter getIsoFile) *getIso {
+	return &getIso{getter: getter}
+}
+
+// GetIsoPluggable defines the method to use get the baseIso file
+var GetIsoPluggable = downloadIso
+
+// Download the ISO using the URL in rhcos.json
+func downloadIso(config *types.InstallConfig) (string, error) {
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	// Get the ISO to use from rhcos.json
+	st, err := rhcos.FetchCoreOSBuild(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// Defaults to using the X86_64 baremetal ISO for all platforms
+	// archName := arch.RpmArch(string(config.ControlPlane.Architecture))
+	archName := "x86_64"
+	streamArch, err := st.GetArchitecture(archName)
+	if err != nil {
+		return "", err
+	}
+	if artifacts, ok := streamArch.Artifacts["metal"]; ok {
+		if format, ok := artifacts.Formats["iso"]; ok {
+			url := format.Disk.Location
+
+			cachedImage, err := DownloadImageFile(url)
+			if err != nil {
+				return "", errors.Wrapf(err, "failed to download base iso image %s", url)
+			}
+			return cachedImage, nil
+		}
+	} else {
+		return "", errors.Wrap(err, "invalid artifact")
+	}
+
+	return "", fmt.Errorf("no iso found to download for %s", archName)
+}
+
+func getIsoFromReleasePayload(config *types.InstallConfig) (string, error) {
+
+	// TODO
+	return "", nil
+}
+
+// Dependencies returns dependencies used by the asset.
+func (i *BaseIso) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
```

Comment:
We can't really depend on the install-config until https://issues.redhat.com/browse/AGENT-136 is implemented, because it will make it compulsory to provide an install-config.

Reply:
Changed to remove InstallConfig as a dependency.